include/hipSYCL/algorithms/numeric.hpp

```diff
@@ -191,6 +191,23 @@ transform_reduce(sycl::queue &q, util::allocation_group &scratch_allocations,
                                        kernel, reduce, deps);
 }
 
+template <class ForwardIt, class T, class Kernel, class BinaryReductionOp,
+          class UnaryTransformOp>
+sycl::event
+find_helper(sycl::queue &q, util::allocation_group &scratch_allocations,
+                 ForwardIt first, ForwardIt last, T *out, T init,
+                 Kernel kernel,
+                 BinaryReductionOp reduce, UnaryTransformOp transform,
+                 const std::vector<sycl::event>& deps = {}) {
+  if(first == last)
+    return sycl::event{};
+
+  std::size_t n = std::distance(first, last);
+
+  return detail::transform_reduce_impl(q, scratch_allocations, out, init, n,
+                                       kernel, reduce, deps);
+}
```

Comment:
`find_helper` here is essentially `transform_reduce`. Since I wanted to adapt the `transform` lambda, I added a separate function instead of modifying the already implemented overloads.

I face the issue that a call to find either returns an uninitialised memory :( or just the first position of the vector. Not sure where I make a mistake, maybe in the call to reducer?